pcsx2/Recording/InputRecording.cpp

```diff
@@ -0,0 +1,205 @@
+#include "PrecompiledHeader.h"
+
+#include "AppSaveStates.h"
+#include "AppGameDatabase.h"
+#include "Common.h"
+#include "Counters.h"
+#include "MemoryTypes.h"
+#include "SaveState.h"
+
+#include "Recording/RecordingControls.h"
+#include "InputRecording.h"
+
+#include <vector>
+
+
+InputRecording g_InputRecording;
+
+// --------------------------------
+// Tag and save framecount along with savestate
+// --------------------------------
+void SaveStateBase::InputRecordingFreeze()
+{
+	FreezeTag("InputRecording");
+	Freeze(g_FrameCount);
+	if (IsLoading()) {
+		g_InputRecordingData.addUndoCount();
+	}
+}
+
+//----------------------------------
+// Main func for handling controller input data
+// Called by Sio.cpp::sioWriteController
+//----------------------------------
+void InputRecording::ControllerInterrupt(u8 &data, u8 &port, u16 & bufCount, u8 buf[])
+{
+	// TODO - Multi-Tap Support
+	// Only examine controllers 1 / 2
+	if (port < 0 || 1 < port)
+		return;
+
+	//==========================
+	// This appears to try to ensure that we are only paying attention
+	// to the frames that matter, the ones that are reading from
+	// the controller.
+	//
+	// See - Lilypad.cpp:1254
+	// 0x42 is the magic number for the default read query
+	//
+	// NOTE: this appears to break if you have logging enabled in LilyPad's config!
```

Comment:
Such a problem should either be propagated where people will actually see it, or resolved.